src/modules/CaaSMapper.ts

```diff
@@ -116,36 +116,51 @@ export class CaaSMapper {
     )
     this.logger = logger
     this.referenceDepth = utils.referenceDepth ?? 0
-    this.maxReferenceDepth = utils.maxReferenceDepth ?? DEFAULT_MAX_REFERENCE_DEPTH
+    this.maxReferenceDepth =
+      utils.maxReferenceDepth ?? DEFAULT_MAX_REFERENCE_DEPTH
 
     this.logger.debug('Created new CaaSMapper')
   }
 
-  addToResolvedReferences(item: MappedCaasItem | CaasApi_Item) {
+  addToResolvedReferences(
+    item: MappedCaasItem | CaasApi_Item,
+    remoteProjectId?: string
```

Comment:
Let's just talk about this a little bit. Why does the remoteProject parameters need to be an additional parameter in every single mehtod?

Reply:
Because the mapper was not written with remoteProjects in mind. Once a Document is fetched from a remote Project, it needs to carry its projectId with it, so that its own references can be resolved against the remote Project as well.

Definetly feels clunky, though i don't know if we can do it better without a thorough restructure.